src/lib.rs

```diff
@@ -0,0 +1,1211 @@
+					Arc::clone(&chain_monitor),
+					Arc::clone(&wallet),
+					Arc::clone(&router),
+					Arc::clone(&logger),
+					Arc::clone(&keys_manager),
+					Arc::clone(&keys_manager),
+					Arc::clone(&keys_manager),
+					user_config,
+					chain_params,
+				)
+			}
+		};
+
+		let channel_manager = Arc::new(channel_manager);
+
+		// Give ChannelMonitors to ChainMonitor
+		for (_blockhash, channel_monitor) in channel_monitors.into_iter() {
+			let funding_outpoint = channel_monitor.get_funding_txo().0;
+			chain_monitor.watch_channel(funding_outpoint, channel_monitor);
+		}
+
+		// Initialize the P2PGossipSync
+		let gossip_sync = Arc::new(P2PGossipSync::new(
+			Arc::clone(&network_graph),
+			None::<Arc<dyn UtxoLookup + Send + Sync>>,
+			Arc::clone(&logger),
+		));
+
+		// Initialize the PeerManager
+		let onion_messenger: Arc<OnionMessenger> = Arc::new(OnionMessenger::new(
+			Arc::clone(&keys_manager),
+			Arc::clone(&keys_manager),
+			Arc::clone(&logger),
+			IgnoringMessageHandler {},
+		));
+		let ephemeral_bytes: [u8; 32] = keys_manager.get_secure_random_bytes();
+		let lightning_msg_handler = MessageHandler {
+			chan_handler: Arc::clone(&channel_manager),
+			route_handler: Arc::clone(&gossip_sync),
+			onion_message_handler: onion_messenger,
+		};
+
+		let cur_time = SystemTime::now()
+			.duration_since(SystemTime::UNIX_EPOCH)
+			.expect("System time error: Clock may have gone backwards");
+		let peer_manager: Arc<PeerManager> = Arc::new(PeerManager::new(
+			lightning_msg_handler,
+			cur_time.as_secs().try_into().expect("System time error"),
+			&ephemeral_bytes,
+			Arc::clone(&logger),
+			IgnoringMessageHandler {},
+			Arc::clone(&keys_manager),
+		));
+
+		// Init payment info storage
+		// TODO: persist payment info to disk
+		let inbound_payments = Arc::new(Mutex::new(HashMap::new()));
+		let outbound_payments = Arc::new(Mutex::new(HashMap::new()));
+
+		// Restore event handler from disk or create a new one.
+		let event_queue = if let Ok(mut f) =
+			fs::File::open(format!("{}/{}", ldk_data_dir, event::EVENTS_PERSISTENCE_KEY))
+		{
+			Arc::new(
+				EventQueue::read(&mut f, Arc::clone(&persister))
+					.expect("Failed to read event queue from disk."),
+			)
+		} else {
+			Arc::new(EventQueue::new(Arc::clone(&persister)))
+		};
+
+		let peer_store = if let Ok(mut f) =
+			fs::File::open(format!("{}/{}", ldk_data_dir, peer_store::PEER_INFO_PERSISTENCE_KEY))
+		{
+			Arc::new(
+				PeerInfoStorage::read(&mut f, Arc::clone(&persister))
+					.expect("Failed to read peer information from disk."),
+			)
+		} else {
+			Arc::new(PeerInfoStorage::new(Arc::clone(&persister)))
+		};
+
+		let running = RwLock::new(None);
+
+		Node {
+			running,
+			config,
+			wallet,
+			tx_sync,
+			event_queue,
+			channel_manager,
+			chain_monitor,
+			peer_manager,
+			keys_manager,
+			network_graph,
+			gossip_sync,
+			persister,
+			logger,
+			scorer,
+			inbound_payments,
+			outbound_payments,
+			peer_store,
+		}
```

Comment:
Consider packing these on a few lines if rustfmt supports that. Probably worth doing in a follow-up later if it will result in a lot of churn.